docs/DeveloperGuide.md

```diff
@@ -4,30 +4,136 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+Ideas and structure for the Developer Guide: https://se-education.org/addressbook-level3/DeveloperGuide.html
+
+Ideas and structure for the User Guide: https://se-education.org/addressbook-level3/UserGuide.html
+
+Additional Packages used: JSON
+
+## Setting up and getting started
 
+
+## Design & implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Add Medication Command
+The add medication command extends from the Command parent class and implements the following operations:
+- execute(MedicationManager, DailyMedicationManager, Ui) - Adds the medication object into the respective medication managers.
+- setMedicineAttributes() - Sets the medicine object's attributes to be added to the medicine managers. 
+- assertionTest(MedicationManager, DailyMedicationManager) - Asserts that medicine has been added to both medication managers.
+- parseStringToValues(Arguments) - Parses string input for medicine quantity and dosage into double type.
```

Comment:
Perhaps add sequence diagram or class diagram here to show the interactions and relationships since this seems to involve quite a lot of classes?